### PR TITLE
Extend Support for Multi-Dimensional Arrays

### PR DESCRIPTION
**Pull request by WAcry**
fixes https://github.com/serilog/serilog/issues/2019

developed based on https://github.com/serilog/serilog/pull/2035

To support multi dimensional arrays

### CODE REVIEW

File: test/Serilog.Tests/Core/LoggerTests.cs [hunk 1]
```diff
@@ -317,6 +317,175 @@ public void NullMessageTemplateParametersDoNotBreakBinding()
         // ReSharper restore StructuredMessageTemplateProblem
     }
 
+    // https://github.com/serilog/serilog/issues/2019
+    [Fact]
+    public void Two_Dimensional_Array_Should_Be_Logger_As_Sequence()
```
Comment (nblumhardt) [inline test/Serilog.Tests/Core/LoggerTests.cs:322]: Nit: `_Should_Be_Logger` => `_Should_Be_Logged` typo in this and several other of the test names.

Naming: tests in this project use PascalCase names, so `TwoDimensionalArrayShouldBeLoggedAsSequence` would be the matching casing here.

**Review by nblumhardt — COMMENTED** [review]
This looks great! Thanks @WAcry (and @sungam3r 👍)

CI is failing with some xUnit analyzer issues which the diff view also shows inline.

Otherwise just some minor naming nitpicks - LGTM otherwise.

File: test/Serilog.Tests/Core/LoggerTests.cs (hunk 1 above)
Comment (WAcry) [reply test/Serilog.Tests/Core/LoggerTests.cs]: Fixed Naming and Linting

Comment (nblumhardt) [reply test/Serilog.Tests/Core/LoggerTests.cs]: Thanks! Merged :+1:

**Pull request merged**